iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerWithEngine.java

```diff
@@ -514,6 +519,75 @@ public void testInsertOverwritePartitionedTable() throws IOException {
     HiveIcebergTestUtils.validateData(table, expected, 0);
   }
 
+  @Test
+  public void testCTASFromHiveTable() {
+    Assume.assumeTrue("CTAS target table is supported only for HiveCatalog tables",
+        testTableType == TestTables.TestTableType.HIVE_CATALOG);
```

Comment:
Why? What is blocking us?

Reply:
As discussed, since the `Catalogs.createTable()` does not register the table in HMS for non-HiveCatalogs, any subsequent SELECTS for the target table wouldn't work.